experimental/parser/diagnostics_internal.go

```diff
@@ -132,107 +132,191 @@ func (e errMoreThanOne) Diagnose(d *report.Diagnostic) {
 const (
 	justifyNone int = iota
 	justifyBetween
-	justifyRight
 	justifyLeft
+	justifyRight
 )
 
 type justified struct {
 	report.Edit
 	justify int
 }
 
+// justify generates suggested edits using justification information.
+//
+// "Justification" is a token-aware operation that ensures that each suggested
+// edit is either:
+//
+// 1. Surrounded on both sides by at least once space. (justifyBetween)
+// 2. Has no whitespace to its left or its right. (justifyLeft, justifyRight)
+//
+// This is necessary in order to generate sensible suggestions in a number of
+// cases.
+//
+// # Case 1: Inserting an equals sign.
```

Comment:
Is this the only case, or were you intending to add more? Also, why is this necessary? I guess you are trying to do something clever with whitespace around it? (Maybe include those details, because "insert equals sign" doesn't clearly explain why this stuff is needed/useful.)